keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,680 @@
+
+The algorithm for align-cpus-by-uncorecache will be implemented to follow the default packed behavior of the existing static CPUManager allocation with the introduction of a uncorecache hierarchy. This means that when a guaranteed container requires CPUs that are equal to or greater than the size of a NUMA or socket, the CPU allocation will behave as usual and schedule the full socket or NUMA. The scheduled CPUs will be subtracted from the quantity of CPUs required for the container.
+
+When/once the required CPUs for a container are smaller than the size of a NUMA, the algorithm will be implemented as follows:
+1. Scan each socket in numerical order. If required CPUs is less than the size of available CPUs on the socket, pick this socket.
+2. Within the chosen socket, scan each NUMA in numerical order. If the required CPUs is less than the size of the available CPUs on the NUMA, pick this NUMA.
+3. Within the chosen NUMA, scan through every uncore cache index in numerical order:
+   - If the required CPUs is greater than or equal the size of the **total** amount of CPUs to an uncore cache, assign the full uncore cache CPUs if they are all **unscheduled** within the group. Subtract the amount of scheduled CPUs from the quantity of required CPUs. 
+     - Continue scanning uncore caches in numerical order and assigning full uncore cache CPU groups until the required quantity is less than the **total** number of CPUs to a uncore cache group.
+   - If the required CPUs is less than the size of the **total** amount of CPUs to an uncore cache, scan through each uncore cache index in numerical order starting from the first index, and assign CPUs if there are enough **available** CPUs within the uncore cache group.
+4. If the required amount of CPUs cannot fit within an uncore cache group and there are enough schedulable CPUs on the node, assign cores in numerical order.
+5. Container will not be scheduled on the node if there are not enough CPUs to satisfy the container requirements. 
+
+This scheduling policy will minimize the distribution of containers across uncore caches to improve performance while still maintaining the default packed logic. The scope will be initially be narrowed to implement uncore cache alignment to the default static scheduling behavior. The table below summarizes future enhancement plans to implement uncore cache alignment to be compatible with the distributed scheduling policies to reduce contention/noisy neighbor effects.
+
+
+| Compatibility | alpha | beta | GA |
+| --- | --- | --- | --- |
+| full-pcpus-only | x | x | x |
+| distribute-cpus-across-numa  |   | x | x |
+| align-by-socket | x | x | x |
+| distribute-cpus-across-cores |   | x | x |
+
+### Test Plan
+
+HW Test Matrix
+
+- 1P 32C (smt-on) NPS=1 //1 numa node
+- 1P32C (smt-off)  NPS=1 // 1 numa numa
+- 2P 128C (smt-on) NPS=1 // 2 numa nodes here  
+- 2P 96C (smt-off) NPS=2  // 4 numa nodes here  
+- 2P Saphire Rapid 
```

Comment:
Please don't use code names in the KEPs. If you want to refer to particular Intel CPU model, use official naming: 
"4th Gen Intel® Xeon® Scalable Processors" (reference https://ark.intel.com/content/www/us/en/ark/products/series/228622/4th-gen-intel-xeon-scalable-processors.html )

Second thing: if you're testing different NPS modes on AMD cpus, consider to also test similar configuration on Intel CPUs. Option called in BIOS: SNC (sub-numa clustering), which is available for several generations that are currently on the market.


Ideally, it would be also good to test implemented algorithm with hybrid cores, like [12th Generation Intel® Core™ i7 Processors](https://ark.intel.com/content/www/us/en/ark/products/series/217837/12th-generation-intel-core-i7-processors.html) which have L2 clustering, similarly to Graviton below.

Reply:
Done, changed names and added hardware to test plan